rocketpy/simulation/flight.py

```diff
@@ -1120,6 +1134,23 @@ def __init_equations_of_motion(self):
             # NOTE: The u_dot is faster, but only works for solid propulsion
             self.u_dot_generalized = self.u_dot
 
+    def __init_controllers(self):
+        """Initialize controllers"""
+        self._controllers = self.rocket._controllers[:]
+        if self._controllers:
+            if self.time_overshoot == True:
+                self.time_overshoot = False
+                warnings.warn(
+                    "time_overshoot has been set to False due to the presence of controllers. "
+                )
+            # reset controllable object to initial state (only airbrakes for now)
+            for air_brakes in self.rocket.air_brakes:
+                air_brakes._reset()
+
+    def __cache_post_process_variables(self):
```

Comment:
What is the benefit of tis `__cache_post_process_variables` method?

Wouldn't it get to the same result if we simply called "np.array(self.__post_processed_variables)" directly?

I'm sorry but I had a hard time understanding all these caches and variables storages.

Reply:
You are right. Right now it changes nothing. I think it was needed before the PR refactoring the post-process

fixed https://github.com/RocketPy-Team/RocketPy/pull/586/commits/464f2017c46367857582b33b81c7379d4d39cd80